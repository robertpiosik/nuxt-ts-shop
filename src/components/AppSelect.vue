<template lang="pug">
	select(v-model="value" @change="input()")
		option(disabled value="") {{ label }}
		option(v-for="(option, index) in options" :key="index" :value="option.value") {{ option.label }}
</template>

<script lang="ts">
import { Vue, Component, Prop, Emit } from 'vue-property-decorator'

@Component
export default class AppSelect extends Vue {
	@Prop({type: String, required: true}) readonly label!: String
	@Prop({ type: Array, required: true }) readonly options!: object[]

	value = ""

	@Emit()
	input() {
		return this.value
	}

}
</script>
