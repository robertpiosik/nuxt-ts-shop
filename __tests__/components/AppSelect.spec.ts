import { mount } from '@vue/test-utils'

import AppSelect from './../../src/components/AppSelect.vue'

const label = 'Select option'

const options = [
  { label: 'Option 1', value: 'option-1' },
  { label: 'Option 2', value: 'option-2' }
]

const wrapper = mount(AppSelect, {
  propsData: {
    label,
    options
  }
})

describe('Component: AppSelect', () => {
  describe('GENERAL', () => {
    it('should match a snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
    it('should be a Vue instance', () => {
      expect(wrapper.isVueInstance()).toBeTruthy()
    })
  })
  describe('PROPS', () => {
    describe('label', () => {
      it('should receive', () => {
        expect(wrapper.props('label')).toBe(label)
      })
    })
    describe('options', () => {
      expect(wrapper.props('options')).toBe(options)
    })
  })
})
