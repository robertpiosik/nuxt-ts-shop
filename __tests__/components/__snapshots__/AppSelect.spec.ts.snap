// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AppSelect GENERAL should match a snapshot 1`] = `
<select class="select">
  <option disabled="disabled" value="">Select option</option>
  <option value="option-1">Option 1</option>
  <option value="option-2">Option 2</option>
</select>
`;
